pkg/tfbridge/detailed_diff.go

```diff
@@ -259,6 +292,78 @@ func (differ detailedDiffer) makeListDiff(
 	return diff
 }
 
+type setChangeIndex struct {
+	engineIndex   int
+	newStateIndex int
+	oldChanged    bool
+	newChanged    bool
+}
+
+func (differ detailedDiffer) makeSetDiff(
+	path propertyPath, old, new resource.PropertyValue,
+) map[detailedDiffKey]*pulumirpc.PropertyDiff {
+	diff := make(map[detailedDiffKey]*pulumirpc.PropertyDiff)
+	oldList := old.ArrayValue()
+	newList := new.ArrayValue()
+	newInputsList := []resource.PropertyValue{}
+
+	newInputs, newInputsOk := path.GetFromMap(differ.newInputs)
+	if newInputsOk && isPresent(newInputs) && newInputs.IsArray() {
+		newInputsList = newInputs.ArrayValue()
+	}
+
+	oldIdentities := differ.calculateSetHashIndexMap(path, resource.NewArrayProperty(oldList))
+	newIdentities := differ.calculateSetHashIndexMap(path, resource.NewArrayProperty(newList))
+	inputIdentities := hashIndexMap{}
+
+	if !schemaContainsComputed(path, differ.tfs, differ.ps) {
```

Comment:
This is code smell to me, care to explain? Very much possible I'm missing a subtlety, but I think this distinction should be avoidable.

Reply:
Yeah, if the schema contains computed then the input is not of the right shape as the output, so hashing might not work, some fields might be missing etc. In those cases we can not correctly match outputs to inputs so we just bail and depend on https://github.com/pulumi/pulumi-terraform-bridge/blob/26d954926f37b55e1ac8591832663165e0f9659c/pkg/tfbridge/detailed_diff.go#L342 to print a warning to the user.

Reply:
I am worried about this one since I think a lot of schemas contain Computed. You mentioned in convo today that hashing might panic. Or it might not work. I think this situation warrants trying to compute the hash and hash-based match anyway, and recovering from panic, on the chance it will actually work. We can backlog. This only makes sense to do if the current code gives up on showing a diff it would otherwise show in some very very common scenario. I have a hunch it might be the case but no proof. We can ship this and revisit later.

Reply:
Discussed offline. This should be good enough for now but we can explore improvements here. Opened https://github.com/pulumi/pulumi-terraform-bridge/issues/2528